lib/tests/streamlit/delta_generator_test.py

```diff
@@ -143,36 +143,42 @@ def test_sidebar_nonexistent_method(self):
 
     @parameterized.expand(
         [
-            (st.empty().empty, "streamlit.delta_generator", "empty", "()"),
-            (st.empty().text, "streamlit.delta_generator", "text", "(body)"),
+            (lambda: st.empty().empty, "streamlit.delta_generator", "empty", "()"),
+            (lambda: st.empty().text, "streamlit.delta_generator", "text", "(body)"),
             (
-                st.empty().markdown,
+                lambda: st.empty().markdown,
                 "streamlit.delta_generator",
                 "markdown",
                 "(body, unsafe_allow_html=False)",
             ),
             (
-                st.empty().checkbox,
+                lambda: st.empty().checkbox,
                 "streamlit.delta_generator",
                 "checkbox",
                 "(label, value=False, key=None)",
             ),
             (
-                st.empty().dataframe,
+                lambda: st.empty().dataframe,
                 "streamlit.delta_generator",
                 "dataframe",
                 "(data=None, width=None, height=None)",
             ),
             (
-                st.empty().add_rows,
+                lambda: st.empty().add_rows,
                 "streamlit.delta_generator",
                 "add_rows",
                 "(data=None, **kwargs)",
             ),
-            (st.write, "streamlit.delta_generator", "write", "(*args, **kwargs)"),
+            (
+                lambda: st.write,
+                "streamlit.delta_generator",
+                "write",
+                "(*args, **kwargs)",
+            ),
         ]
     )
-    def test_function_signatures(self, func, module, name, sig):
+    def test_function_signatures(self, lambda_func, module, name, sig):
+        func = lambda_func()
```

Comment:
Curious, what changed so that we had to wait to execute st.empty() inside of the test function?

Reply:
Contexts are added on setUp, and the parameterized are executed before the setUp call, so it raised an exception. This worked, but then Cypress failed on executing bare scripts. Reverting back because the suggested solution should fix it.